internal/resources/google/compute_vpn_gateway.go

```diff
@@ -0,0 +1,39 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+)
+
+type ComputeVpnGateway struct {
+	Address string
+	Region  string
+
+	MonthlyEgressDataTransferGB *ComputeVPNGatewayNetworkEgressUsage `infracost_usage:"monthly_egress_data_transfer_gb"`
+}
+
+var ComputeVPNGatewayUsageSchema = []*schema.UsageItem{
+	{
+		Key:          "monthly_egress_data_transfer_gb",
+		ValueType:    schema.SubResourceUsage,
+		DefaultValue: &usage.ResourceUsage{Name: "monthly_egress_data_transfer_gb", Items: ComputeVPNGatewayNetworkEgressUsageSchema},
+	},
+}
+
+func (r *ComputeVpnGateway) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *ComputeVpnGateway) BuildResource() *schema.Resource {
+	region := r.Region
+	r.MonthlyEgressDataTransferGB.Region = region
```

Comment:
Should we check for nil here to prevent nil pointer possibilities?

Reply:
Yeah, I made a mistake in adding the checks in another function. Moved them all in their own function.